output/client.go

```diff
@@ -161,27 +187,40 @@ func (logzioClient *LogzioClient) createRequest() (*http.Request, int) {
 func (logzioClient *LogzioClient) doRequest(req *http.Request) int {
 	resp, err := logzioClient.client.Do(req)
 	if err != nil {
-		logzioClient.logger.Log(fmt.Sprintf("failed to do client request: %+v", err))
-		return output.FLB_ERROR
+		logzioClient.logger.Log(fmt.Sprintf("failed to do client request (retryable): %+v", err))
+		return output.FLB_RETRY
 	}
+
 	defer resp.Body.Close()
 
+	// While we should be able to read the response body, it's not required.  so log but don't return
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		logzioClient.logger.Log(fmt.Sprintf("received an error attempting to read from logz.io listener: %+v", err))
-		return resp.StatusCode
+		logzioClient.logger.Log(fmt.Sprintf("failed attempting to read from logz.io listener: %+v.  Status %v", err, resp.StatusCode))
 	}
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		logzioClient.logger.Log(fmt.Sprintf("received a non-2xx HTTP status code from logz.io listener: %d (%v)", resp.StatusCode, string(body)))
-		return resp.StatusCode
+	// retry in the same way as the fluent bit http plugin
+	if resp.StatusCode >= 500 {
+		logzioClient.logger.Log(fmt.Sprintf("received retryable HTTP status code from logz.io listener: %d", resp.StatusCode))
+		if body != nil {
+			logzioClient.logger.Log(fmt.Sprintf("  (%v)", string(body)))
+		}
+		return output.FLB_RETRY
+	} else if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+		logzioClient.logger.Log(fmt.Sprintf("received non-retryable HTTP status code from logz.io listener: %d", resp.StatusCode))
+		if body != nil {
+			logzioClient.logger.Log(fmt.Sprintf("  (%v)", string(body)))
+		}
+		return output.FLB_ERROR
```

Comment:
Having this logic here, makes function shouldRetry redundent.
I think this logic should move to shouldRetry, and in doRequest just to return the status code.
I this function I think we should do - if not 2xx, return output.FLB_ERROR.